### ISSUE TITLE
require-returns-check: False positive for function returning function

### ISSUE THREAD

**Issue by ilyub** [gajus/eslint-plugin-jsdoc#934]
## Expected behavior

Should pass without errors

## Actual behavior

Getting error:
```
JSDoc @returns declaration present but return expression not available
```

## ESLint Config

```js
rules: { "jsdoc/require-returns-check": "warn" }
```

## ESLint sample

```ts
export const f =
  /**
   * Description.
   *
   * @returns Result.
   */
  () => {
    return () => {};
  };
```

## Environment

- Node version: 18.12.1
- ESLint version 8.28.0
- `eslint-plugin-jsdoc` version: 39.6.2

**Comment by brettz9** [comment]
The top comment block was mistakenly being found by the inner function as though that were its return statement. Surprising this hadn't been caught before. Thanks for the PR!
